File: ObjC/BookshelfObjC/Book.m
//
//  Book.m
//  BookshelfObjC
//
//  Created by Victor Barros on 2016-03-04.
//  Copyright © 2016 Kinvey. All rights reserved.
//

#import "Book.h"

@implementation Book

+(NSString *)kinveyCollectionName
{
    return @"Book";
}

+(NSDictionary<NSString *,NSString *> *)kinveyPropertyMapping
{
    return @{@"objectId" : KNVPersistableIdKey,
             @"title" : @"title",
             @"publicationDate" : @"publication_date"};
}

@end

File: Swift/Bookshelf/Book.swift
//
//  Book.swift
//  Bookshelf
//
//  Created by Victor Barros on 2016-02-08.
//  Copyright © 2016 Kinvey. All rights reserved.
//

import Foundation
import Kinvey

class Book: Entity {
    
    dynamic var title: String?
    
    override static func collectionName() -> String {
        return "Book"
    }
    
    override func propertyMapping(map: Map) {
        super.propertyMapping(map)
        
        title <- map["title"]
    }

}
